File: proxy-pattern/src/main/java/com/example/proxy_pattern/service/CacheEntry.java
package com.example.proxy_pattern.service;

import com.example.proxy_pattern.dto.ReportGeneratedResponseDto;

public class CacheEntry {
    private final ReportGeneratedResponseDto response;
    private final long timestamp;

    public CacheEntry(ReportGeneratedResponseDto response, long timestamp) {
        this.response = response;
        this.timestamp = timestamp;
    }

    public ReportGeneratedResponseDto getResponse() {
        return response;
    }

    public long getTimestamp() {
        return timestamp;
    }
}

File: proxy-pattern/src/services/CacheEntry.ts
import { ReportGeneratedResponseDto } from '../dtos/ReportGeneratedResponseDto';

/**
 * Represents a cache entry for report generation responses.
 */
export class CacheEntry {
  /**
   * Creates a new CacheEntry.
   * @param response The report generation response.
   * @param timestamp The timestamp when the entry was created.
   */
  constructor(
    private readonly response: ReportGeneratedResponseDto,
    private readonly timestamp: number
  ) {}

  /**
   * Gets the report generation response.
   * @returns The report generation response.
   */
  public getResponse(): ReportGeneratedResponseDto {
    return this.response;
  }

  /**
   * Gets the timestamp when the entry was created.
   * @returns The timestamp.
   */
  public getTimestamp(): number {
    return this.timestamp;
  }
}